torch/csrc/distributed/rpc/python_functions.cpp

```diff
@@ -136,9 +139,12 @@ c10::intrusive_ptr<JitFuture> wrapFutureMessageInJitFuture(
   } else {
     c10::intrusive_ptr<JitFuture> jitFuture =
         c10::make_intrusive<JitFuture>(NoneType::get());
-
+    // Save and pass thread local state into the callback
+    at::ThreadLocalState tls_state;
```

Comment:
Seems this can be moved out of the if-else clause, as both cases require this tls?

Reply:
Fixed, for the other cases as well